pennylane/qnode.py

```diff
@@ -427,6 +427,15 @@ def __init__(
                     f"gradient kwargs."
                 )
 
+        if mode is None:
+            mode = "best"
+        else:
+            warnings.warn(
+                "The `mode` keyword argument is deprecated and does nothing with the new return system. "
+                "Please use `grad_on_execution` instead.",
+                UserWarning,
+            )

```

Comment:
Minor point, but should this be raised when the old return system is enabled?

Reply:
Honestly, I'm a little confused as to why we didn't have a warning about a keyword argument we were just going to completely ignore.  I think the warning needs to be here.

Reply:
I'm just not sure that it's the right warning if a user is working with the old return system. We tell them "use `grad_on_execution` instead", but this is overidden by `mode` in the old return system as @lillian542 [mentioned](https://github.com/PennyLaneAI/pennylane/pull/4316#discussion_r1254518065).

When `not qml.active_return()`, we could either have:
1. Not have a warning at all. Users will have already seen a warning when calling `qml.disable_return()`.
2. Have this warning be slightly different, instead directing users to again try out the new return system.